### ISSUE TITLE
Undefined variable 'nextnode'

### ISSUE THREAD

**Issue by firda-cze** [KSP-KOS/KOS#2237]
I was really trying to reduce this, but I could not. Try running `control.ks`, add one manuever node (or two), execute one by pressing `EXEC` which runs `node.ks` and it will do its job, press that again (while having node) and it will crash with very surprising **`Undefined Variable Name 'nextnode'`. WTF?! That is built-in!**

[kos.zip](https://github.com/KSP-KOS/KOS/files/1776458/kos.zip)

It points to `node.ks:6: parameter nodeEta is { return nextNode:eta. }.`
How can `nextNode` exist the first time and not the second time?!

EDIT: Removed unnecessary code for `circ.ks` and `meetn.ks`.
EDIT2: Try reduced kos3.zip from https://github.com/KSP-KOS/KOS/issues/2237#issuecomment-370602471

**Comment by firda-cze** [comment]
I was able to fix this by moving everything to functions. The problem seems to be in execution of programs/scripts.

[kos2.zip](https://github.com/KSP-KOS/KOS/files/1776581/kos2.zip)

**Comment by brantwedel** [comment]
I think it is that at the end of your original `node.ks` file you are removing nextNode `if hasNode remove nextNode.` ... so by the time the function gets called later ... there is no nextnode

**Comment by firda-cze** [comment]
@brantwedel No, I said make two nodes or one after executig first. None of the scrips is creating any node, you have to create it yourself. It fails having node.

**Comment by firda-cze** [comment]
I have just reproduced the error again by downloading my own zip from this issue report, `run control.ks`, adding two nodes and altering their prograde (of first) and retrograde (of second) and executing first and then executing the second. Please try it. I know some things do not always work the same, but first try it before imagining my idiocy, please.

**Comment by brantwedel** [comment]
Sorry, I reproduced it as well, I was too quick on the reply, I should have run the code / looked deeper before commenting. (I'm not part of the KOS team btw, just trying to help out)

**Comment by firda-cze** [comment]
[kos3.zip](https://github.com/KSP-KOS/KOS/files/1782942/kos3.zip)

Reduced it to a point, where removing few lines of code makes it working.
`reduced.ks` is still crashing, but `working.ks` is trully working. The difference is removal of these lines:

```
local row is lhs:addHLayout().
local dirPro is radio(dir,button(row,"Prograde",100)).
local dirRet is radio(dir,button(row,"Retrograde",100)).
```
and
```
bindToggle(dirPro, dirToggle, { sas off. lock steering to prograde. }).
bindToggle(dirRet, dirToggle, { sas off. lock steering to retrograde. }).
```

I have no idea, those two buttons are never pressed, but `set dirUnl:pressed to true.` calls `dirToggle` which is `onToggle` with `dir` and `dirUnl` bound, and that calls `uncheck(dir,dirUnl)` which contains
`for btn in lst if btn <> except set btn:pressed to false.`

In working version, there is no `btn <> except` because they are both `dirUnl` and there is no other button, but in crashing version, this calls `set btn:pressed to false.` for those two buttons (`dirPro` and `dirRet`). How could that make a difference?

P.S.: Either use `bindToggle` from original zip, or my fix #2238 for the other bug with bound arguments in a trigger.

**Comment by MoritzGoeckel** [comment]
I encountered the exact same problem. I was executing subprograms with run "xx.ks"(arguments). The first time it worked, but when I execute it the second time, it will break in the LOCK complaining about "undefined variable nextnode". Tried many different ways, the lock would just always break, even though the variable definitely existed (I checkd with the "defined" keyword).

Same as op wrote: It worked for me after I made the subprogramm file a function and only executed it once. I was then able to execute the code as often as necessary via the function call without any issues.

**TLDR: Had same problem, it seems to persist. Same solution as ops worked.**
